libethash/CMakeLists.txt

```diff
@@ -37,6 +37,8 @@ if (CRYPTOPP_FOUND)
 	TARGET_LINK_LIBRARIES(ethash ${CRYPTOPP_LIBRARIES})
 endif()
 
-TARGET_LINK_LIBRARIES(ethash m)
+if (CMAKE_COMPILER_IS_GNUCC)
```

Comment:
I think it does, because the issue is in GCC 6 but not in previous versions. Can someone test it?

In case of clang, remember to add this for linux only.